include/rocksdb/utilities/cache_dump_load.h

```diff
@@ -0,0 +1,82 @@
+//  Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved.
+//  This source code is licensed under both the GPLv2 (found in the
+//  COPYING file in the root directory) and Apache 2.0 License
+//  (found in the LICENSE.Apache file in the root directory).
+
+#pragma once
+
+#include <set>
+
+#include "rocksdb/cache.h"
+#include "rocksdb/env.h"
+#include "rocksdb/file_system.h"
+#include "rocksdb/io_status.h"
+#include "rocksdb/secondary_cache.h"
+#include "rocksdb/table.h"
+
+namespace ROCKSDB_NAMESPACE {
+
+static const int kCacheDumpMajorVersion = 0;
+static const int kCacheDumpMinorVersion = 1;
+
+class CacheDumpWriter {
+ public:
+  virtual ~CacheDumpWriter() = default;
+
+  // Called ONCE before the calls to WritePacket
+  virtual IOStatus WriteMetadata(const Slice& metadata) = 0;
+  virtual IOStatus WritePacket(const Slice& data) = 0;
+  virtual IOStatus Close() = 0;
+};
+
+class CacheDumpReader {
+ public:
+  virtual ~CacheDumpReader() = default;
+  // Called ONCE before the calls to ReadPacket
+  virtual IOStatus ReadMetadata(std::string* metadata) = 0;
+  // Sets data to empty string on EOF
+  virtual IOStatus ReadPacket(std::string* data) = 0;
+  // (Close not needed)
+};
+
+// CacheDumpOptions is used to control DumpCache()
+struct CacheDumpOptions {
+  SystemClock* clock;
+};
+
+class CacheDumper {
```

Comment:
I don't understand the point of abstract classes for CacheDumper and CacheDumpedLoader. I don't see a future in which we call these from RocksDB, as I believe the intention is for the user to call them. So why provide an interface where users can override some functionality that only they would be calling themselves? Unnecessary API is confusing IMHO. Why not just provide functions & options for what we support? (Options object of course can be expanded in the future for supporting more variations on the same functionality, or more functions added.)

Reply:
At the beginning, I thought that maybe user will follow the public abstract class to implement their own dump/load version. But you are right, in general, cache dump/load is an independent tool. If user want to have their own version, they can just implement one without using any interface from here. I will remove the abstract classes.